datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt

```diff
@@ -1130,15 +1153,15 @@ internal class QuestionnaireViewModel(application: Application, state: SavedStat
       // Results in a new questionnaire state being generated synchronously, i.e., the current
       // thread will be suspended until the new state is generated.
       modificationCount.update { it + 1 }
-      forceValidation = false
-    }
 
-    if (
-      currentPageItems.filterIsInstance<QuestionnaireAdapterItem.Question>().all {
-        it.item.validationResult is Valid
+      viewModelScope.launch {
+        _questionnaireStateFlow.take(1).collectLatest {
+          forceValidation = false
+          checkPageItemsAllValid(it.items)
+        }
```

Comment:
consider using `Flow<T>.first()` since we're only expecting the first emission
https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-core/common/src/flow/terminal/Reduce.kt#L82-L94

```suggestion
        checkPageItemsAllValid(_questionnaireStateFlow.first().items)
        forceValidation = false
```

n.b. it's important to have forceValidation after the state call